pip/basecommand.py

```diff
@@ -314,6 +314,23 @@ def populate_requirement_set(requirement_set, args, options, finder,
                     'You must give at least one requirement to %(name)s '
                     '(see "pip help %(name)s")' % opts)
 
+        # On Windows, any operation modifying pip should be run as:
+        #     python -m pip ...
+        # See https://github.com/pypa/pip/issues/1299 for more discussion
+        should_show_use_python_msg = (
+            sys.platform == 'win32' and
```

Comment:
Jython trunk, which will be the 2.7.1 release:

```
>>> import os, platform
>>> platform.python_implementation()
'Jython'
```

On OS X:
```
>>> platform.java_ver()
('1.8.0_112', 'Oracle Corporation', ('Java HotSpot(TM) 64-Bit Server VM', '25.112-b16', 'Oracle Corporation'), ('Mac OS X', '10.12.4', 'x86_64'))
>>> os.name
'java ( ==posix for targets )'
>>> os.name == 'java'
True
>>> os._name
'posix'
```

On Windows:
```
>>> platform.python_implementation( )
'Jython'
>>> os.name
'java ( ==nt for targets )'
>>> os._name
'nt'
```

On Ubuntu 16.04:
```
>>> os.name
'java ( ==posix for targets )'
>>> os._name
'posix'
>>> platform.python_implementation()
'Jython'
>>> platform.java_ver()
('1.7.0_80', 'Oracle Corporation', ('Java HotSpot(TM) 64-Bit Server VM', '24.80-b11', 'Oracle Corporation'), ('Linux', '4.4.0-66-generic', 'amd64'))
```

We can stop here, and just suggest some combination of the data above to determine the correct functionality is the right way to do this.

But you are a Python enthusiast, so it's worth pointing out that `os.name` has grown some functionality with Jython 2.7.1, as described in http://bugs.jython.org/issue2557 - this is the targets functionality, and it's why `os.name == 'java'` is `True` above, despite the `__repr__` being `'java ( ==posix for targets )'`. This is implemented by providing a specific target string implementation in Jython. (Fortunately adding new builtin types in Jython is rather cheap.)


```
>>> type(os.name)
<type 'shadowstr'>
```

You might think this would allow for immediate comparison of `os.name == 'posix'`, but the Python ecosystem is just too rich in its complexification for something so simple like that to work; in this case I'm trying this out on OSX:

```
>>> os.name == 'posix'
False
```

Instead, it's necessary to register such interest, via `os.name.addtarget('org\.python.*')`, which is global; or via a scheme that enables for this registration to be done from a distance. See the above Jython bug for more details.